### PR TITLE
refactor!: move experimental avatar to kit

### PR DESCRIPTION
**Pull request by splincode**
Closes #

### CODE REVIEW

**Review by waterplea — APPROVED** [review]

File: projects/demo/src/modules/app/pages.ts [hunk 1]
```diff
@@ -159,8 +159,8 @@ export const pages: TuiDocPages = [
     },
     {
         section: 'Components',
-        title: 'Avatar',
-        keywords: 'аватар, image, pic, картинка, изображение, avatar, stack',
+        title: 'Avatar ',
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/app/pages.ts]: fixed

File: projects/demo/src/modules/components/avatar/examples/import/import-module.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ```ts
 import {NgModule} from '@angular/core';
-import {TuiAvatarModule} from '@taiga-ui/kit';
+import {TuiAvatarModule} from '@taiga-ui/kot';
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/avatar/examples/import/import-module.md]: fixed

File: projects/demo/src/modules/components/combo-box/examples/1/index.html [hunk 3]
```diff
@@ -65,9 +65,10 @@
         <tui-avatar
             size="xs"
             class="avatar"
-            [avatarUrl]="data.avatarUrl || null"
-            [text]="data.toString()"
-        ></tui-avatar>
+            [src]="data.avatarUrl || null"
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/combo-box/examples/1/index.html]: fixed

File: projects/demo/src/modules/components/checkbox-block/examples/3/index.html [hunk 4]
```diff
@@ -10,10 +10,11 @@
             <div class="content">
                 <tui-avatar
                     size="s"
-                    text="Heading"
                     class="tui-space_right-3"
-                    [rounded]="true"
-                ></tui-avatar>
+                    [round]="true"
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/checkbox-block/examples/3/index.html]: fixed

File: projects/demo/src/modules/components/input/input.module.ts [hunk 5]
```diff
@@ -86,6 +87,7 @@ import {ExampleTuiInputComponent} from './input.component';
         TuiTextCodeModule,
         MaskitoDirective,
         TuiUnmaskHandlerModule,
+        TuiAutoColorModule,
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/input/input.module.ts]: fixed

File: projects/demo/src/modules/components/block-status/examples/4/index.html [hunk 6]
```diff
@@ -10,10 +10,10 @@
             <tui-avatar
                 *ngFor="let user of users"
                 size="l"
-                [autoColor]="true"
-                [rounded]="true"
-                [text]="user"
-            ></tui-avatar>
+                [style.background]="user | tuiAutoColor"
+            >
+                {{ user | tuiInitials }}
```
Comment (waterplea) [inline projects/demo/src/modules/components/block-status/examples/4/index.html:15]: It is meant to be `[src]="user | tuiInitials"` as per design font is a bit bigger that way.

File: projects/demo/src/modules/components/select/examples/6/index.html [hunk 7]
```diff
@@ -21,9 +21,10 @@
         <tui-avatar
             size="xs"
             class="avatar"
-            [avatarUrl]="data.avatarUrl || null"
-            [text]="data.toString()"
-        ></tui-avatar>
+            [src]="data.avatarUrl"
```
Comment (waterplea) [inline projects/demo/src/modules/components/select/examples/6/index.html:24]: And such examples are probably meant to be `[src]="data.avatarUrl || (data.toString() | tuiInitials)"`

File: projects/demo/src/modules/components/sheet-dialog/sheet-dialog.template.html [hunk 8]
```diff
@@ -57,11 +57,7 @@
             </button>
             <ng-template #template>
                 <div class="label">
-                    <tui-avatar
-                        size="m"
-                        text="Karl Gambolputty"
-                        [rounded]="true"
-                    ></tui-avatar>
+                    <tui-avatar size="m">Karl Gambolputty</tui-avatar>
```
Comment (waterplea) [inline projects/demo/src/modules/components/sheet-dialog/sheet-dialog.template.html:60]: This one would overflow. One way to handle it, other than initials, is `<span tuiFade>...</span>`

File: projects/kit/components/avatar/avatar.component.ts [hunk 9]
```diff
@@ -1,93 +1,75 @@
...
-
-    @Input()
-    autoColor: boolean = this.options.autoColor;
-
-    @Input()
-    @HostBinding('class._rounded')
-    rounded: boolean = this.options.rounded;
-
-    avatarUrl: SafeResourceUrl | string | null = null;
+    src?: SafeResourceUrl | string | null = null;
```
Comment (waterplea) [inline projects/kit/components/avatar/avatar.component.ts:48]: nit: no need for initial `= null` now

Comment (splincode) [reply projects/kit/components/avatar/avatar.component.ts]: fixed

File: projects/demo/src/modules/components/sheet-dialog/sheet-dialog.template.html (hunk 8 above)
Comment (splincode) [reply projects/demo/src/modules/components/sheet-dialog/sheet-dialog.template.html]: fixed

File: projects/demo/src/modules/components/block-status/examples/4/index.html (hunk 6 above)
Comment (splincode) [reply projects/demo/src/modules/components/block-status/examples/4/index.html]: fixed

**Pull request merged**